design/ironic-interaction/ironic-background-docs.md

```diff
@@ -0,0 +1,346 @@
+optional and can be provided/updated later, so in theory the minimal json
+request is (using ipmi driver as example):
+
+    POST /v1/nodes
+    {"driver": "ipmi"}
+
+Valid hardware and maintained in-tree hardware types, or drivers, as of
+ironic 12.0.0 are: idrac, ilo, ipmi, irmc, redfish, snmp, and xclarity.
+
+Usually more info are provided, at least a node name and parameters to
+initialize the drivers, such as username and password, if needed, passed
+through the “driver_info” option.
+
+A more complete example of a standard node create request in json format:
+
+    {
+    "name": "test_node_dynamic",
+    "driver": "ipmi",
+    "driver_info": {
+        "ipmi_username": "ADMIN",
+        "ipmi_password": "password"
+    },
+    "power_interface": "ipmitool"
+    }
+
+The response, if successful, contains a complete record of the node in json
+format with provided or default ({},“null”, or “”) values.
+
+## Updating information about a hardware node in ironic.
+
+All node information can be updated after the node has been created.
+
+Send a PATCH to /v1/nodes/node-name in the form of a JSON PATCH document.
+
+The normal response is a 200 and contains a complete record of the node in
+json format with the updated node.
+
+### How to inspect hardware?
+
+Inspection may be used if a baremetal node has not been already discovered
+or inspected previously in order to collect up to date details about the
+hadware. This is particullarlly important in order to update the records
+of networking ports for identification of the baremetal node and creation
+of PXE/iPXE configuration files in order to help ensure that baremetal node
+is quickly booted for booting into the deployment and discovery ramdisk.
+
+    PUT /v1/nodes/node-name/provision/states
+    {"target": "inspect"}
+
+This operation can only be performed in the "manageable" state. If the
+node is already in the "available" state, the same requst can be used
+with a target of "manage", and then the target of "inspect" can be utilized
+to step through the state machine.
+See: https://docs.openstack.org/ironic/latest/contributor/states.html
+
+After inspection, it is advisable to return nodes to an "available" state.
+This can be performed simillarlly via the target "provide".
+
+### How to deploy?
+
+Starting with the bare metal node in the "available" provision_state:
+
+1. First assert configuration to API to indicate the the image written
+   to disk. This is performed as a HTTP PATCH request to the
+   /v1/nodes/node-name endpoint.
+
+       {
+       {“op”: “replace”, “path”: “/instance_info”, “value”: {
+           “image_source”: “http://url-to-image”,
+           “image_os_hash_algo”: “sha256”,
+           “image_os_hash_value”: “abcdefghi…”}},
+       {“op”: “replace”, “path”: “/instance_uuid”, “value”: “anyuuidvalue”}},
+       }
+
+   **NOTE:** Instead of defining the "image_os_hash_*" values, a MD5 based
+   image checksum can be set.
```

Comment:
Are the hash parameters required?

Reply:
a checksum of some sort is required, or are you attempting to clarify if they are required at all?